src/main/java/com/hivemq/cli/commands/options/TlsOptions.java

```
@@ -138,104 +127,255 @@ private boolean useBuiltSslConfig() {
+     *
+     * @return a key manager factory or null if no key provider was configured.
+     * @throws Exception if a keystore or private key could not be accessed.
+     */
+    private @Nullable KeyManagerFactory buildKeyManagerFactory() throws Exception {
+        final KeyManagerFactory keyManagerFactory;
+
         final DefaultCLIProperties defaultCLIProperties =
                 Objects.requireNonNull(MqttCLIMain.MQTTCLI).defaultCLIProperties();
-
-        if (clientCertificateChain == null) {
+        final Path defaultKeystore = defaultCLIProperties.getKeystore();
+        final Path defaultClientCertificateChain = defaultCLIProperties.getClientCertificateChain();
+        final Path defaultClientPrivateKey = defaultCLIProperties.getClientPrivateKey();
+
+        final boolean keystoreExists = clientKeystorePath != null;
+        final boolean onlyCertificateExists = clientCertificatePath != null && clientPrivateKeyPath == null;
+        final boolean onlyPrivateKeyExists = clientCertificatePath == null && clientPrivateKeyPath != null;
+        final boolean certificateAndPrivateKeyExists = clientCertificatePath != null && clientPrivateKeyPath != null;
+        final boolean defaultKeystoreExists = defaultKeystore != null;
+        final boolean onlyDefaultCertificateExists =
+                defaultClientCertificateChain != null && defaultClientPrivateKey == null;
+        final boolean onlyDefaultPrivateKeyExists =
+                defaultClientCertificateChain == null && defaultClientPrivateKey != null;
+        final boolean defaultCertificateAndPrivateKeyExists =
+                defaultClientCertificateChain != null && defaultClientPrivateKey != null;
+
+        if (keystoreExists) {
+            if (certificateAndPrivateKeyExists) {
+                Log.warn("The keystore parameter shadows the private key corresponding certificate parameters. " +
+                        "If you want to use them instead please remove the keystore parameter.");
```

I would throw an Exception here.